### PR TITLE
Attempts to fix all physics bugs at once

### LINKED ISSUE #3591: Fix collision masks (related to items)
Now that items actually have collision a bunch of stuff is colliding with them that shouldn't (e.g. dispensers, etc.)

At round start there were ~160 contacts being processed for collision so someone needs to go through these and fix them.

The general solution is make these have MobImpassable layers instead of Impassable so items can still go through them.

### PR DESCRIPTION
**Pull request by SweptWasTaken**
## About the PR 

This PR attempts to close #3591.

It also does some minor cleaning up and QOL stuff for Constructible related entities.

**Draft PR** for now so @metalgearsloth can inspect the work so far and see if it's the right path. Sorry for clogging it with some unnecessary stuff but I had to fix them or I would explode.

The stuff touching collisions in components is on the advice of sloth.

**Screenshots**

Nothing really interesting in this PR so here's me with a plant bumping into my back.

**Changelog**

:cl:
- add: You can anchor and rotate a lot more things
- tweak; Fixed physics for a lot of things (hopefully)
- remove: Some unused furniture sprites

### CODE REVIEW

File: Resources/Prototypes/Entities/Constructible/Furniture/carpets.yml [hunk 1]
```diff
@@ -1,26 +1,22 @@
...
   - type: Physics
     fixtures:
-      - shape:
-          !type:PhysShapeAabb
-            bounds: "-0.5, -0.5, 0.5, 0.5"
-        layer: [ Passable ]
+    - shape:
+        !type:PhysShapeAabb {}
+      layer:
+      - Passable
```
Comment (PaulRitter) [inline Resources/Prototypes/Entities/Constructible/Furniture/carpets.yml:19]: invalid yaml

File: Resources/Prototypes/Entities/Constructible/Furniture/instruments.yml [hunk 2]
```diff
@@ -35,6 +25,7 @@
     interfaces:
     - key: enum.InstrumentUiKey.Key
       type: InstrumentBoundUserInterface
+  - type: Climbable
```
Comment (PaulRitter) [inline Resources/Prototypes/Entities/Constructible/Furniture/instruments.yml:28]: make instruments climbable? ok i guess?

File: Resources/Prototypes/Entities/Constructible/Furniture/instruments.yml [hunk 3]
```diff
@@ -69,17 +58,4 @@
...
-  name: xylophone
-  parent: BasePlaceableInstrument
-  id: XylophoneInstrument
-  description: Rainbow colored glockenspiel.
-  components:
-  - type: Instrument
-    program: 13
-  - type: Sprite
-    sprite: Objects/Fun/Instruments/otherinstruments.rsi
-    state: xylophone
```
Comment (PaulRitter) [inline Resources/Prototypes/Entities/Constructible/Furniture/instruments.yml:85]: and the xylophone is yet to be seen again

File: Resources/Prototypes/Entities/Constructible/Furniture/storage.yml [hunk 4]
```diff
@@ -1,28 +1,29 @@
 - type: entity
   id: Rack
   name: rack
-  description: "A rack for storing things on."
+  description: A rack for storing things on.
   placement:
     mode: SnapgridCenter
   components:
   - type: Clickable
-  - type: InteractionOutline
```
Comment (PaulRitter) [inline Resources/Prototypes/Entities/Constructible/Furniture/storage.yml:9]: p sure you'd still want an interactionoutline on this one

**Review by PaulRitter — CHANGES_REQUESTED** [review]
jfc look at your diff swept, what is this webedit-tier yaml?

File: Resources/Prototypes/Entities/Constructible/Furniture/instruments.yml (hunk 3 above)
Comment (SweptWasTaken) [reply Resources/Prototypes/Entities/Constructible/Furniture/instruments.yml:85]: I changed it to be handheld.

File: Resources/Prototypes/Entities/Constructible/Furniture/instruments.yml (hunk 2 above)
Comment (SweptWasTaken) [reply Resources/Prototypes/Entities/Constructible/Furniture/instruments.yml]: Yeah, only meant for the piano though. I updated it.

File: Resources/Prototypes/Entities/Constructible/Furniture/potted_plants.yml [hunk 5]
```diff
@@ -1,32 +1,25 @@
...
-      - MobImpassable
-      - VaultImpassable
-      - SmallImpassable
       layer:
-      - Opaque
-      - MobImpassable
-      - VaultImpassable
       - SmallImpassable
+      mask:
+      - VaultImpassable
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Constructible/Furniture/potted_plants.yml:17]: Shouldn't the mask be Impassable?

File: Resources/Prototypes/Entities/Constructible/Furniture/potted_plants.yml [hunk 6]
```diff
@@ -59,11 +52,16 @@
...
-          !type:PhysShapeAabb
-            bounds: "-0.47,-0.25,0.05,0.25"
-        layer: [ Passable ]
+    - shape:
+        !type:PhysShapeAabb
+          bounds: "-0.47,-0.25,0.05,0.25"
+      layer:
+      - SmallImpassable
+      mask:
+      - VaultImpassable
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Constructible/Furniture/potted_plants.yml:64]: Same here

File: Resources/Prototypes/Entities/Objects/base.yml [hunk 7]
```diff
@@ -23,8 +23,8 @@
       - Clickable
       mask:
       - Impassable
-      - SmallImpassable
-      restitution: 0.3  # We want them items B O U N C Y # No we don't
+      - VaultImpassable
+      restitution: 0.05 # RIP Bouncy 2021-2021
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/base.yml:27]: 2nd opinion, also not just making it 0 is stupid.

File: Resources/Prototypes/Entities/Constructible/base.yml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  - type: Physics
+    bodyType: Dynamic
+    mass: 50
+    fixtures:
+    - shape:
+        !type:PhysShapeAabb {}
+      layer:
+      - SmallImpassable
+      mask:
+      - VaultImpassable
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Constructible/base.yml:43]: Impassable

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by SweptWasTaken** [comment]
reviews applied less gooo!!

image

**Comment by SweptWasTaken** [comment]
Build is failing because SharedPlayerMobMoverComponent.cs got a class that uses ICollideSpecial that has since been removed. Up to @metalgearsloth to review this one.

**Review by metalgearsloth — APPROVED** [review]

**Comment by metalgearsloth** [comment]
TableDummy test is failing.

**Comment by SweptWasTaken** [comment]
> 
> 
> TableDummy test is failing.

Fixed, but idk about the rest.

**Review by metalgearsloth — APPROVED** [review]
Probably okay now

**Pull request merged**